### ISSUE TITLE
building shows lots of warnings

### ISSUE THREAD

**Issue by tcurdt** [bitstadium/HockeySDK-iOS#343]
I am building the SDK with carthage
```
github "bitstadium/HockeySDK-iOS" "4.1.1"
```

which gives:
```
$ carthage update --platform iOS
*** Downloading HockeySDK-iOS.framework binary at "Version 4.1.1"
*** Checking out HockeySDK-iOS at "4.1.1"
*** xcodebuild output can be found in /var/folders/pf/7vhqx5bn41qddypw08w9jc4w0000gn/T/carthage-xcodebuild.sIYCU5.log
*** Building scheme "HockeySDK Framework" in HockeySDK.xcodeproj

The following commands produced analyzer issues:
  AnalyzeShallow /project/Carthage/Checkouts/HockeySDK-iOS/Classes/BITFeedbackListViewController.m
  AnalyzeShallow /project/Carthage/Checkouts/HockeySDK-iOS/Classes/BITUpdateViewController.m
  AnalyzeShallow HockeySDKTests/BITChannelTests.m
(3 commands with analyzer issues)

The following commands produced analyzer issues:
  AnalyzeShallow /project/Carthage/Checkouts/HockeySDK-iOS/Classes/BITFeedbackListViewController.m
  AnalyzeShallow /project/Carthage/Checkouts/HockeySDK-iOS/Classes/BITUpdateViewController.m
  AnalyzeShallow HockeySDKTests/BITChannelTests.m
(3 commands with analyzer issues)
/project/Carthage/Checkouts/HockeySDK-iOS/Classes/BITFeedbackListViewController.m:1014:3: warning: Null is returned from a method that is expected to return a non-null value
/project/Carthage/Checkouts/HockeySDK-iOS/Classes/BITUpdateViewController.m:499:3: warning: Null is returned from a method that is expected to return a non-null value
/project/Carthage/Checkouts/HockeySDK-iOS/Support/HockeySDKTests/BITChannelTests.m:31:27: warning: Null is assigned to a pointer which is expected to have non-null value
warning: /var/folders/g9/mxr37xcs33g38q39zkf76xb00000gp/C/com.apple.DeveloperTools/7.2-7C68/Xcode/SharedPrecompiledHeaders/OCMockLib-Prefix-aguhwhqcwfsriffwgjqlwwuendhr/OCMockLib-Prefix.pch.pch: No such file or directory
warning: /var/folders/g9/mxr37xcs33g38q39zkf76xb00000gp/C/com.apple.DeveloperTools/7.2-7C68/Xcode/SharedPrecompiledHeaders/OCMockLib-Prefix-efhbsnkhmearojbasgiitukrjnoo/OCMockLib-Prefix.pch.pch: No such file or directory
/project/Carthage/Checkouts/HockeySDK-iOS/Classes/BITFeedbackListViewController.m:1014:3: warning: Null is returned from a method that is expected to return a non-null value
/project/Carthage/Checkouts/HockeySDK-iOS/Classes/BITUpdateViewController.m:499:3: warning: Null is returned from a method that is expected to return a non-null value
/project/Carthage/Checkouts/HockeySDK-iOS/Support/HockeySDKTests/BITChannelTests.m:31:27: warning: Null is assigned to a pointer which is expected to have non-null value
warning: /var/folders/g9/mxr37xcs33g38q39zkf76xb00000gp/C/com.apple.DeveloperTools/7.2-7C68/Xcode/SharedPrecompiledHeaders/OCMockLib-Prefix-efeadqrgtnidxwbadwtohuxejrdb/OCMockLib-Prefix.pch.pch: No such file or directory
warning: /var/folders/g9/mxr37xcs33g38q39zkf76xb00000gp/C/com.apple.DeveloperTools/7.2-7C68/Xcode/SharedPrecompiledHeaders/OCMockLib-Prefix-bxixbexkmoooetbthjtvimnkelbz/OCMockLib-Prefix.pch.pch: No such file or directory
```

**Comment by TroubleMakerBen** [comment]
Hi Thorsten,
thanks for getting in touch. We'll improve that for 4.1.2 (no ETA, yet, sorry). We wanted to include fixes for the warnings in 4.1.1 but with the Xcode 8 and iOS 10 changes at last minute, they just didn't make it.

Have a great weekend!

Benny

**Comment by TroubleMakerBen** [comment]
Fixed as of 4.1.2.
